__tests__/app.test.js

```diff
@@ -118,6 +108,105 @@ describe("/api/articles/:article_id", () => {
+            expect(article_id).toBe(1);
+            expect(typeof comment_id).toBe("number");
+            expect(typeof votes).toBe("number");
+            expect(typeof created_at).toBe("string");
+            expect(typeof author).toBe("string");
+            expect(typeof body).toBe("string");
+          }
+        );
+        expect(comments).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
+  test("GET 200: Responds with an empty array when article has no comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then(({ body }) => {
+        const { comments } = body;
+        expect(comments).toHaveLength(0);
+      });
+  });
+  test("GET 404: Responds with an appropriate status and error message when given a non-existent article id", () => {
+    return request(app)
+      .get("/api/articles/14/comments")
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("article does not exist");
+      });
+  });
+  test("GET 400: Responds with an appropriate status and error message when given an invalid article id", () => {
+    return request(app)
+      .get("/api/articles/not-an-article/comments")
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("bad request");
+      });
+  });
+  test("POST 201: Responds with the posted comment for an article", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is eloquently crafted.",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment).toMatchObject({
+          comment_id: 19,
+          body: "This is eloquently crafted.",
+          article_id: 1,
+          author: "butter_bridge",
+          votes: 0,
+        });
+        expect(typeof comment.created_at).toBe("string");
+      });
+  });
+  test("POST 404: Responds with an appropriate status and error message when given a non-existent article id", () => {
+    const newComment = {
+      username: "butter_bridge",
+      body: "This is eloquently crafted.",
+    };
+    return request(app)
+      .post("/api/articles/14/comments")
+      .send(newComment)
+      .expect(404)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("article does not exist");
+      });
+  });
+  test("POST 400: Responds with an appropriate status and error message when provided with a bad body (no body)", () => {
+    const newComment = {
+      username: "butter_bridge",
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("bad request");
+      });
+  });
+});
```

Comment:
A great start here, some other errors to consider:
- posting to an invalid article id
- posting with a username that doesn't exist in the database
- posting with no comment at all eg {}